### ISSUE TITLE
Cannot delete objects.

### ISSUE THREAD

**Issue by skx** [skx/sos#4]
> Something, _something_, immutable...

We can either declare this a design choice, or we can add support for it in **both** the upload-server and the blob-server.

**Comment by skx** [comment]
FWIW I did implement this in the blob-server layer, but it was doomed because it would only delete from the specific blob-server.

Consider the case where you have two blob servers, replicated:

* Content stored on `server-A`, in `dataA/`
* Content stored on `server-B`, in `dataB/`

If the delete command hits the second server the data is removed.  When replication is executed it will be recopied from the valid copy on `server-A`.

In short to handle delete we _either_ need to:

* Ensure the API-server polls all back-ends
  * Not an option, we have to assume some are done.
* Record "deleted" statuses of objects, somehow.
  * Currently file with id "`ID`" is written as `ID` + `ID.json`.
  * We could remove `ID`, and `ID.json`, and create `ID.deleted` as a flag
  * But the whole thing is more annoying than it needs to be.
